Sources/SwiftSyntax/Syntax.swift

```diff
@@ -196,105 +161,89 @@ extension Syntax {
   /// The absolute position of the starting point of this node. If the first token
   /// is with leading trivia, the position points to the start of the leading
   /// trivia.
-  public var position: AbsolutePosition {
+  var position: AbsolutePosition {
     return data.position
   }
 
   /// The absolute position of the starting point of this node, skipping any
   /// leading trivia attached to the first token syntax.
-  public var positionAfterSkippingLeadingTrivia: AbsolutePosition {
+  var positionAfterSkippingLeadingTrivia: AbsolutePosition {
     return data.positionAfterSkippingLeadingTrivia
   }
 
   /// The absolute position where this node (excluding its trailing trivia)
   /// ends.
-  public var endPosition: AbsolutePosition {
+  var endPosition: AbsolutePosition {
```

Comment:
I feel like, if these are going to become internal, we might as well get rid of them and just dispatch through `data` inside the module. What do you think?

Reply:
I assume you mean all four position-related vars. I don't have a particular objection but could you clarify what is the benefit of removing them ?

Reply:
Just because they don't serve much purpose, and they're an extra hop. They'll likely be inlined anyway, but still.

Reply:
I realized a subtle problem with removing properties from `_SyntaxBase` and only have them on `Syntax`.
The point of mirroring the `Syntax` API over `_SyntaxBase` is so that internally we can call the APIs and avoid the unnecessary checks that the object conforms to `_SyntaxBase`.
But if an API exists only on `Syntax` and not on `_SyntaxBase` then it is easy to inadvertently call it internally. For example if there is code such as `node.position` where `node` is a `_SyntaxBase`, by removing `position` from `_SyntaxBase` it makes the call go through `Syntax` and the unnecessary check.